src/ImageSharp/Processing/Transforms/Transform.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Numerics;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using SixLabors.ImageSharp.Processing.Processors;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Extension methods for the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static partial class ImageExtensions
+    {
+        /// <summary>
+        /// Transforms an image by the given matrix.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to transform.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <returns>The <see cref="Image{TPixel}"/></returns>
+        public static IImageProcessingContext<TPixel> Transform<TPixel>(this IImageProcessingContext<TPixel> source, Matrix3x2 matrix)
+            where TPixel : struct, IPixel<TPixel>
+            => Transform(source, matrix, KnownResamplers.NearestNeighbor);
+
+        /// <summary>
+        /// Transforms an image by the given matrix using the specified sampling algorithm.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to transform.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <param name="sampler">The <see cref="IResampler"/> to perform the resampling.</param>
+        /// <returns>The <see cref="Image{TPixel}"/></returns>
+        public static IImageProcessingContext<TPixel> Transform<TPixel>(this IImageProcessingContext<TPixel> source, Matrix3x2 matrix, IResampler sampler)
+            where TPixel : struct, IPixel<TPixel>
+            => Transform(source, matrix, sampler, Size.Empty);
+
+        /// <summary>
+        /// Transforms an image by the given matrix using the specified sampling algorithm
+        /// and a rectangle defining the transform origin in the source image and the size of the result image.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to transform.</param>
+        /// <param name="matrix">The transformation matrix.</param>
+        /// <param name="sampler">The <see cref="IResampler"/> to perform the resampling.</param>
+        /// <param name="rectangle">
+        /// The rectangle defining the transform origin in the source image, and the size of the result image.
```

Comment:
@JimBobSquarePants I renamed it back to `rectangle`, I hope my comments explain it better now. If still unsatisfied & there is no better idea, we can internalize this overload for now. 
The one with the `destinationSize` parameter is that really matters.

Reply:
Yeah that's better. 👍 I'll update the submodule and get this merged.